tests/integration-tests/tests/storage/test_efs.py

```diff
@@ -68,6 +74,129 @@ def test_efs_same_az(region, scheduler, pcluster_config_reader, clusters_factory
+    efs_template.set_description("EFS stack created for testing existing EFS")
+    efs_template.add_resource(FileSystem("FileSystemResource"))
+    efs_stack = CfnStack(
+        name="integ-tests-efs-{0}{1}{2}".format(
+            random_alphanumeric(),
+            "-" if request.config.getoption("stackname_suffix") else "",
+            request.config.getoption("stackname_suffix"),
+        ),
+        region=region,
+        template=efs_template.to_json(),
+    )
+    cfn_stacks_factory.create_stack(efs_stack)
+
+    # write file stack contains a mount target and a instance to write a empty file with random name into the efs
+    write_file_template = Template()
+    write_file_template.set_version("2010-09-09")
+    write_file_template.set_description("Stack to write a file to the existing EFS")
+    default_security_group_id = (
+        boto3.client("ec2", region_name=region)
+        .describe_security_groups(
+            Filters=[
+                {"Name": "vpc-id", "Values": [vpc_stack.cfn_outputs["VpcId"]]},
+                {"Name": "group-name", "Values": ["default"]},
+            ]
+        )
+        .get("SecurityGroups")[0]
+        .get("GroupId")
+    )
+    write_file_template.add_resource(
+        MountTarget(
+            "MountTargetResource",
+            FileSystemId=efs_stack.cfn_resources["FileSystemResource"],
+            SubnetId=vpc_stack.cfn_outputs["PublicSubnetId"],
+            SecurityGroups=[default_security_group_id],
+        )
+    )
+    efs_stack.random_file_name = random_alphanumeric()
+    user_data = (
+        """
+    #cloud-config
+    package_update: true
+    package_upgrade: true
+    runcmd:
+    - yum install -y amazon-efs-utils
+    - yum install -y nfs-utils
+    - file_system_id_1="""
+        + efs_stack.cfn_resources["FileSystemResource"]
+        + """
+    - efs_mount_point_1=/mnt/efs/fs1
+    - mkdir -p "${efs_mount_point_1}"
+    - mount -t efs ${file_system_id_1}:/ ${efs_mount_point_1}
+    - touch ${efs_mount_point_1}/"""
+        + efs_stack.random_file_name
+    )
+    write_file_template.add_resource(
+        Instance(
+            "InstanceToWriteEFS",
+            ImageId=retrieve_latest_ami(region, "alinux2"),
+            InstanceType="c5.xlarge",
+            SubnetId=vpc_stack.cfn_outputs["PublicSubnetId"],
+            UserData=Base64(user_data),
+            KeyName=key_name,
+            DependsOn=["MountTargetResource"],
+        )
+    )
+    write_file_stack = CfnStack(
+        name="integ-tests-efs-write-file-{0}{1}{2}".format(
+            random_alphanumeric(),
+            "-" if request.config.getoption("stackname_suffix") else "",
+            request.config.getoption("stackname_suffix"),
+        ),
+        region=region,
+        template=write_file_template.to_json(),
+    )
+    cfn_stacks_factory.create_stack(write_file_stack)
+    logging.info("Sleeping for 30 seconds for the instance to execute user data")
+    time.sleep(30)
```

Comment:
Instead of sleeping what about checking the status of the stack and then of the instance? I don't think that 30 seconds are enough to create the stack, start the instance and execute the code.

Let's use something similar to `utils.verify_stack_creation`

Reply:
Solved by using `cfn-signal`
The 30 seconds were not for creating the instance. They were for running the `userdata`